### ISSUE TITLE
IllegalArgumentException removeProperty(key) for not out-unique properties

### ISSUE THREAD

**Issue by gjdev** [thinkaurelius/titan#290]
The api-docs on AbstractVertex.removeProperty(key) suggest that calling removeProperty(key) for a not out-unique (i.e. multivalued) property key, will delete all values and return the last deleted. The actual implementation throws an IllegalArgumentException instead.

Either the doc or the implementation should be changed. My preference would be to fix the implementation, as it's very useful to be able to delete all values of a vertex for some key.

**Comment by dalaro** [comment]
I agree.  I don't see any implementation or conceptual reason that this method can't behave as the interface describes.  I added a test to verify the behavior described in the interface and then I changed the implementation of `AbstractVertex.removeProperty(TitanKey)` to comply: 48c680b16582318630c56716b0ffc629177841db.  `AbstractVertex.removeProperty(String)` is implemented atop the `TitanKey` equivalent and hence is also affected.
